File: src/countingValleys.py
def countingValleys(n, s):
    level = 0
    levels = [0]
    valleys = 0
    for step in s:
        if step == 'U':
            level += 1
        elif step == 'D':
            level -= 1
        else:
            print("Invalid Input")
            break
        levels.append(level)

    for i in range(len(levels)-1):
        if levels[i]==0 and levels[i+1]==-1:
            valleys += 1
    return valleys

File: src/countingValleys.scala
object Solution {

    // Complete the countingValleys function below.
    def countingValleys(n: Int, s: String): Int = {

        var level = 0
        for (step <- s) {
            if (step == "U") {
                
            }
        }

    }


def countingValleys(n: Int, s: String): Int = s
  .map(_ match {case 'U' => +1; case 'D' => -1})
  .scan(0)(_+_)
  .sliding(2)
  .count(_ == Vector(-1, 0))